llvm/lib/Transforms/Scalar/LoopStrengthReduce.cpp

```diff
@@ -947,7 +947,8 @@ static Immediate ExtractImmediate(const SCEV *&S, ScalarEvolution &SE) {
                            SCEV::FlagAnyWrap);
     return Result;
   } else if (EnableVScaleImmediates)
-    if (const SCEVMulExpr *M = dyn_cast<SCEVMulExpr>(S))
+    if (const SCEVMulExpr *M = dyn_cast<SCEVMulExpr>(S);
+        M && M->getNumOperands() == 2)
```

Comment:
Up to you but personally I think this would be cleaner and more consistent with the surrounding code if flipped like:

```
else if (const SCEVMulExpr *M = dyn_cast<SCEVMulExpr>(S))
  if (EnableVScaleImmediates && M->getNumOperands() == 2)
.....
````

```

Reply:
Sure, makes sense :+1: